frontend/app/profile/edit/page.tsx

```diff
@@ -0,0 +1,279 @@
+        headers: {
+          "Content-Type": "multipart/form-data",
+        },
+      })
+      .then((response) => {
+        if (response.status === 200) {
+          toast({
+            variant: "success",
+            title: "Success",
+            description: "Profile updated.",
+          });
+          router.push("/profile/" + user?.userId);
+        }
+      })
+      .catch((error: AxiosError) => {
+        let title = "Error";
+        let message = error.message;
+
+        if (error.response) {
+          switch (error.response.status) {
+            case 400:
+              message =
+                "The request was invalid. Please check your input and try again.";
+              break;
+            case 401:
+              message = "You are not authorized to edit this profile.";
+              break;
+            case 403:
+              message = "You do not have permission to perform this action.";
+              break;
+            case 404:
+              message = "The profile you are trying to edit does not exist.";
+              break;
+            case 500:
+              message =
+                "An internal server error occurred. Please try again later.";
+              break;
+            case 503:
+              message =
+                "The service is currently unavailable. Please try again later.";
+              break;
+            default:
+              message = "An unexpected error occurred. Please try again.";
+          }
+        }
+
+        showErrorToast(title, message);
+      });
+  };
+
+  const handleChange = (
+    e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
+  ) => {
+    const { name, value, files } = e.target as HTMLInputElement;
+    if (name === "profilePicturePath" && files?.[0]) {
+      const file = files[0];
+
+      // Validate the image file
+      const validation = validateImageFile(file);
+      if (!validation.isValid) {
+        showErrorToast("Invalid File", validation.error ?? "Invalid file");
+        return;
+      }
+
+      const imageUrl = URL.createObjectURL(file);
+      setSelectedImage(imageUrl);
+      setPlayerProfileRequest((prev) => ({
+        ...prev,
+        profileUpdates: {
+          ...prev.profileUpdates,
+          profilePicturePath: imageUrl,
+        },
+        profileImage: file,
+      }));
+    } else {
+      setPlayerProfileRequest((prev) => ({
+        ...prev,
+        profileUpdates: {
+          ...prev.profileUpdates,
+          [name]: value || null,
+        },
+      }));
+    }
+  };
+
+  const handleClearImage = () => {
+    setSelectedImage(null);
+    setPlayerProfileRequest((prev) => ({
+      ...prev,
+      profileUpdates: {
+        ...prev.profileUpdates,
+        profilePicturePath:
+          PROFILE_IMAGE_API + playerProfileRequest.profileUpdates.username,
+      },
+      profileImage: new File([], ""),
+    }));
+  };
+
+  if (isLoading) return <div>Loading...</div>;
```

Comment:
i might do a loading animation screen and we can just insert the animation screen